### PR TITLE
Fortigate update

### PR DESCRIPTION
**Pull request by ayman-m**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Description
- Fixed an issue with authentication in case of Fortigate authentication disclaimer configured.
- Added the "fortigate-create-address" command to create a new address object.
- Added the "fortigate-delete-address" command to delete an existing address object.

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x ] No

## Must have
- [ ] Tests
- [x ] Documentation

### CODE REVIEW

File: Packs/FortiGate/Integrations/FortiGate/FortiGate.py [hunk 1]
```diff
@@ -1149,6 +1153,110 @@ def delete_address_group_request(name):
...
 
+@logger
+def create_address_command():
+    contents = []
+    context = {}
+    address_context = []
+    address_name = demisto.args().get('name', '')
+    address = demisto.args().get('address', '')
+    mask = demisto.args().get('mask', '')
+    fqdn = demisto.args().get('fqdn', '')
```
Comment (MosheEichler) [inline Packs/FortiGate/Integrations/FortiGate/FortiGate.py:1164]:
```suggestion
    args = demisto.args()
    address_name = args.get('name', '')
    address = args.get('address', '')
    mask = args.get('mask', '')
    fqdn = args.get('fqdn', '')
```

File: Packs/FortiGate/Integrations/FortiGate/FortiGate.py [hunk 2]
```diff
@@ -1149,6 +1153,110 @@ def delete_address_group_request(name):
...
+
+    if address:
+        contents.append({
+            'Name': address_name,
+            'IPAddress': address
+        })
+        address_context.append({
+            'Name': address_name,
+            'IPAddress': address
+        })
```
Comment (MosheEichler) [inline Packs/FortiGate/Integrations/FortiGate/FortiGate.py:1179]:
```suggestion
        address_dict = {
            'Name': address_name,
            'IPAddress': address
        }
        contents.append(address_dict)
        address_context.append(address_dict)
```

File: Packs/FortiGate/Integrations/FortiGate/FortiGate.py [hunk 3]
```diff
@@ -1149,6 +1153,110 @@ def delete_address_group_request(name):
...
+    elif fqdn:
+        contents.append({
+            'Name': address_name,
+            'FQDN': fqdn
+        })
+        address_context.append({
+            'Name': address_name,
+            'FQDN': fqdn
+        })
```
Comment (MosheEichler) [inline Packs/FortiGate/Integrations/FortiGate/FortiGate.py:1188]:
```suggestion
        fqdn_dict = {
            'Name': address_name,
            'FQDN': fqdn
        }
        contents.append(fqdn_dict)
        address_context.append(fqdn_dict)
```

File: Packs/FortiGate/Integrations/FortiGate/FortiGate.py [hunk 4]
```diff
@@ -1149,6 +1153,110 @@ def delete_address_group_request(name):
...
+def create_address_request(address_name, address, mask, fqdn):
+    uri_suffix = 'cmdb/firewall/address/'
+    if does_path_exist(uri_suffix + address_name):
+        return_error('Address already exists.')
+    subnet = address + " " + mask
+    if address:
+        payload = {
+            'name': address_name,
+            'subnet': subnet
+        }
```
Comment (MosheEichler) [inline Packs/FortiGate/Integrations/FortiGate/FortiGate.py:1212]:
```suggestion
    if address:
        subnet = address + " " + mask
        payload = {
            'name': address_name,
            'subnet': subnet
        }
```

File: Packs/FortiGate/Integrations/FortiGate/FortiGate.py [hunk 5]
```diff
@@ -1149,6 +1153,110 @@ def delete_address_group_request(name):
...
+    delete_address_request(name)
+
+    contents.append({
+        'Name': name,
+        'Deleted': True
+    })
+    address_context.append({
+        'Name': name,
+        'Deleted': True
+    })
```
Comment (MosheEichler) [inline Packs/FortiGate/Integrations/FortiGate/FortiGate.py:1239]:
```suggestion
    address_dict = {
        'Name': name,
        'Deleted': True
    }
    contents.append(address_dict)
    address_context.append(address_dict)
```

**Review by MosheEichler — CHANGES_REQUESTED** [review]
HI [ayman-m](https://github.com/ayman-m),
You did a fantastic job :)

Please see my comments
In addition, the build fails please update from master.
Ant let me know when you are done.

Thanks!

File: Packs/FortiGate/Integrations/FortiGate/README.md [hunk 6]
```diff
@@ -1186,3 +1186,126 @@
 <p> </p>
 <h5>Command Example</h5>
 <pre>  !fortigate-get-banned-ips </pre>
+<h3 >19. Create an address object</h3>
```
Comment (MosheEichler) [inline Packs/FortiGate/Integrations/FortiGate/README.md:1189]: Please add a link to this command in the beginning of the file (lines 44)
Screen Shot 2022-08-01 at 10 30 52

File: Packs/FortiGate/Integrations/FortiGate/README.md [hunk 7]
```diff
@@ -1186,3 +1186,126 @@
...
+<td style="width: 322px;">The address deletion status.</td>
+</tr>
+</tbody>
+</table>
+<p> </p>
+<h5>Command Example</h5>
+<pre>!fortigate-delete-address name=test</pre>
+<h5>Context Example</h5>
+<pre>Fortigate:{} 1 item<br>Address:[] 1 item<br>0:{} 2 items<br>1:{} 2 items<br>Deleted:True<br>Name:test</pre>
+<h5>Human Readable Output</h5>
```
Comment (MosheEichler) [inline Packs/FortiGate/Integrations/FortiGate/README.md:1311]: What is the human readable output?

File: Packs/FortiGate/Integrations/FortiGate/README.md [hunk 8]
```diff
@@ -1186,3 +1186,126 @@
...
+<td style="width: 322px;">The domain name.</td>
+</tr>
+</tbody>
+</table>
+<p> </p>
+<h5>Command Example</h5>
+<pre>!fortigate-create-address name=test address="1.1.1.1"</pre>
+<h5>Context Example</h5>
+<pre>Fortigate:{} 1 item<br>Address:[] 1 item<br>0:{} 2 items<br>1:{} 2 items<br>IPAddress:1.1.1.1/255.255.255.255<br>Name:test</pre>
+<h5>Human Readable Output</h5>
```
Comment (MosheEichler) [inline Packs/FortiGate/Integrations/FortiGate/README.md:1259]: What is the human readable output?

File: Packs/FortiGate/Integrations/FortiGate/README.md [hunk 9]
```diff
@@ -1186,3 +1186,126 @@
...
+<td style="width: 113px;">Required</td>
+</tr>
+<tr>
+<td style="width: 193px;">address</td>
+<td style="width: 402px;">The IP Address, example: 1.1.1.1</td>
+<td style="width: 113px;">Optional</td>
+</tr>
+<tr>
+<td style="width: 193px;">mask</td>
+<td style="width: 402px;">The address mask,example: 255.255.255.0 , Default is 255.255.255.255</td>
```
Comment (MosheEichler) [inline Packs/FortiGate/Integrations/FortiGate/README.md:1216]:
```suggestion
<td style="width: 402px;">The address mask, example: 255.255.255.0 , Default is 255.255.255.255</td>
```

**Review by MosheEichler — CHANGES_REQUESTED** [review]
and few more commanets....

**Comment by ayman-m** [comment]
> HI [ayman-m](https://github.com/ayman-m), You did a fantastic job :)
> 
> Please see my comments In addition, the build fails please update from master. Ant let me know when you are done.
> 
> Thanks!

Hi @MosheEichler , thanks for taking the time reviewing this PR, I have pushed  new commits after reviewing your comments,  I have also  updated from master

**Comment by MosheEichler** [comment]
Hi @ayman-m
I think a lot of files that don't belong were accidentally put into PR.
Please take a look

**Comment by ayman-m** [comment]
> Hi @ayman-m I think a lot of files that don't belong were accidentally put into PR. Please take a look

It looks like something  went wrong while updating  from  master,  will  close this  PR and open a  new one

**Pull request closed without merging**